File: Source/AlphaTab/Rendering/Glyphs/GhostParenthesisGlyph.cs
using AlphaTab.Platform;

namespace AlphaTab.Rendering.Glyphs
{
    class GhostParenthesisGlyph : Glyph
    {
        private readonly bool _isOpen;
        private const int Size = 6;
        public GhostParenthesisGlyph(bool isOpen) : base(0, 0)
        {
            _isOpen = isOpen;
        }

        public float Height { get; set; }

        public override void DoLayout()
        {
            base.DoLayout();
            Width = Size * Scale;
        }

        public override void Paint(float cx, float cy, ICanvas canvas)
        {
            if (_isOpen)
            {
                TieGlyph.PaintTie(canvas, Scale,
                    cx + X + Width, cy + Y + Height,
                    cx + X + Width, cy + Y,
                    false, Size, 3);
            }
            else
            {
                TieGlyph.PaintTie(canvas, Scale,
                    cx + X, cy + Y,
                    cx + X, cy + Y + Height,
                    false, Size, 3);

            }
            canvas.Fill();
        }
    }
}

File: src/rendering/glyphs/GhostParenthesisGlyph.ts
import type { Color } from '@src/model/Color';
import type { ICanvas } from '@src/platform/ICanvas';
import { Glyph } from '@src/rendering/glyphs/Glyph';
import { TieGlyph } from '@src/rendering/glyphs/TieGlyph';

export class GhostParenthesisGlyph extends Glyph {
    private static readonly Size: number = 6;
    private _isOpen: boolean;

    public colorOverride?: Color;

    public constructor(isOpen: boolean) {
        super(0, 0);
        this._isOpen = isOpen;
    }

    public override doLayout(): void {
        super.doLayout();
        this.width = GhostParenthesisGlyph.Size;
    }

    public override paint(cx: number, cy: number, canvas: ICanvas): void {
        const c = canvas.color;
        if (this.colorOverride) {
            canvas.color = this.colorOverride;
        }
        if (this._isOpen) {
            TieGlyph.paintTie(
                canvas,
                1,
                cx + this.x + this.width,
                cy + this.y + this.height,
                cx + this.x + this.width,
                cy + this.y,
                false,
                6,
                3
            );
        } else {
            TieGlyph.paintTie(canvas, 1, cx + this.x, cy + this.y, cx + this.x, cy + this.y + this.height, false, 6, 3);
        }
        canvas.color = c;
    }
}
